app/paypal.py

```
@@ -0,0 +1,231 @@
+                padding.PKCS1v15(),
+                hashes.SHA256(),
+            )
+            return True
+        return False
+    except InvalidSignature:
+        return False
+    except Exception as e:
+        log_exception(e, "verify_paypal_signature")
+        return False
+
+async def get_paypal_public_key(cert_url: str) -> PublicKey | None:
```

Ruff (0.12.2)

Undefined name `PublicKey`

(F821)